src/librustc/mir/mod.rs

```diff
@@ -832,13 +836,56 @@ pub enum StatementKind<'tcx> {
     Nop,
 }
 
+#[derive(Copy, Clone, RustcEncodable, RustcDecodable, PartialEq, Eq)]
+pub enum ValidationOp {
+    Acquire,
```

Comment:
Can we document at least a bit the semantics of each of these operations in doc comment form?